src/vopr.zig

```diff
@@ -597,9 +597,14 @@ pub const Simulator = struct {
             if (simulator.cluster.replica_health[replica_index] == .down) {
                 simulator.restart_replica(@intCast(replica_index), fault);
             }
+            // Disable storage faults during liveness mode.
+            simulator.cluster.storages[replica_index].transition_to_liveness_mode();
         }
 
+        // Disable network faults during liveness mode.
         simulator.cluster.network.transition_to_liveness_mode(simulator.core);
+
+        // Disable process crash/restart faults during liveness mode.
```

Comment:
These comments feel like they don't actually add anything to the code

Reply:
Mhmmmm, I had added them to save a level of indirection. When I first skimmed through this code, I wasn't able to understand the overall picture of liveness mode without peeking into the `network.transition_to_liveness_mode` function. Perhaps better to document this above the function definition instead, then?

Reply:
This probably means that we are missing the overall comment on the `transition_to_liveness_mode`, rather the comment on the impl details. `network.transition_to_liveness_mode` is somewhat intricate, as it doesn't _just_ disables faults, but actually "freezes" existing partitions.

Worth linking https://tigerbeetle.com/blog/2023-07-06-simulation-testing-for-liveness at least!

Reply:
https://github.com/tigerbeetle/tigerbeetle/pull/2334/commits/c8fb7969dd83219cd83cd1366c5e09af543feafc